### ISSUE TITLE
Support for Rcpp::exceptions without call stacks

### ISSUE THREAD

**Issue by jimhester** [RcppCore/Rcpp#664]
I would like to have an API for constructing Rcpp exceptions without call stacks, also potentially also an argument to `Rcpp::stop()` analogous to `stop(call. = FALSE)`, although the latter is mostly just sugar if you can throw the desired Rcpp exception manually.

You can view a working implementation of this is at https://github.com/RcppCore/Rcpp/compare/master...jimhester:exceptions_without_callstacks

**Comment by eddelbuettel** [comment]
So when I call `stop()` I get something like

```r
R> cppFunction('void loudStop() { stop("Boom"); }')
R> loudStop()
Error in loudStop() : Boom
R> 
R> anotherLayer <- function() loudStop()
R> anotherLayer()
Error in loudStop() (from #1) : Boom
R> 
```

Do I understand you correctly that you just want to see `Boom` ?

**Comment by jimhester** [comment]
Yes exactly, this is R equivalent

```r
(function() stop("boom"))()
#> Error in (function() stop("boom"))() : boom

(function() stop("boom", call. = FALSE))()
#> Error: boom
```

**Comment by eddelbuettel** [comment]
I'm in favor.  I am a somewhat-recent (a year? two) convert to `call.=FALSE` which I use more and more.

**Comment by coatless** [comment]
This issue ticket can be closed as it was addressed in #663.
